dll/cpl/desk/settings.c

```diff
@@ -842,6 +844,37 @@ SwitchDisplayMode(HWND hwndDlg, PWSTR DeviceName, PSETTINGS_ENTRY seInit, PSETTI
     }
 }
 
+static
+PSETTINGS_ENTRY
+FindBestElement(
+    _In_ PSETTINGS_ENTRY Request,
+    _In_ PSETTINGS_ENTRY List)
+{
+    LONG Distance, NearestDistance = MAXLONG;
+    PSETTINGS_ENTRY pBestEntry = NULL, Current;
+
+    /* Find the best entry in the list */
+    for (Current = List; Current; Current = Current->Flink)
+    {
+        Distance = 0x100000 * labs(Current->dmBitsPerPel       - Request->dmBitsPerPel      ) +
+                      0x100 * labs(Current->dmPelsWidth        - Request->dmPelsWidth       ) +
+                      0x100 * labs(Current->dmPelsHeight       - Request->dmPelsHeight      ) +
+                              labs(Current->dmDisplayFrequency - Request->dmDisplayFrequency);
+        if (Distance == 0)
+            return Current;
+
+        if (Distance < NearestDistance)
+        {
+            pBestEntry = Current;
+            NearestDistance = Distance;
+        }
+    }
+
+    assert(pBestEntry);
```

Comment:
Not needed anymore, as I assume `List` cannot be empty, can it?

Reply:
Deleted.